manager/orchestrator/update/updater.go

```diff
@@ -520,10 +521,41 @@ func (u *Updater) removeOldTasks(ctx context.Context, batch *store.Batch, remove
 	return removedTask, nil
 }
 
+// Checks if the current assigned node matches the Placement.Constraints
+// specified in the task spec for Updater.newService.
+func (u *Updater) nodeMatches(t *api.Task) bool {
+	if t.Spec.Placement == nil {
+		return true
+	}
+
+	constraints, _ := constraint.Parse(u.newService.Spec.Task.Placement.Constraints)
```

Comment:
Why toss the error? Is it guaranteed that the constraint will parse correctly? If so, can you add a comment to that effect?

Reply:
FWIW I believe there are other places where we ignore the error. The idea is that it's validated elsewhere, and if it's somehow invalid, there's nothing that can be done here except spamming the logs, which we probably don't want to do. If parsing fails, `constraints` will be empty, which is the fallback behavior we would want anyway.